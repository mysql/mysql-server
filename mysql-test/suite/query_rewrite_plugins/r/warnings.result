#
# Test of rules where the parser gives warnings about the originial
# query, pattern or replacement.
#
call mtr.add_suppression("Plugin rewriter reported: 'Some rules failed to load.'");
# Query rewrite plugin was installed.
INSERT INTO query_rewrite.rewrite_rules ( pattern, replacement )
VALUES ( 'LOAD DATA INFILE "tab1.data" INTO TABLE test.t1',
'LOAD DATA INFILE "tab2.data" INTO TABLE test.t1' ),
( 'DO CONCAT("a", "b")', 'DO CONCAT("c", "d")' );
CALL query_rewrite.flush_rewrite_rules();
ERROR 45000: Loading of some rule(s) failed.
SELECT * FROM query_rewrite.rewrite_rules;
id	pattern	pattern_database	replacement	enabled	message
1	LOAD DATA INFILE "tab1.data" INTO TABLE test.t1	NULL	LOAD DATA INFILE "tab2.data" INTO TABLE test.t1	YES	Statement type of pattern not supported.
2	DO CONCAT("a", "b")	NULL	DO CONCAT("c", "d")	YES	Statement type of pattern not supported.
INSERT INTO query_rewrite.rewrite_rules ( pattern, replacement )
VALUES ( 'SHOW WARNINGS', 'SHOW /* I rewrote it */ WARNINGS' );
CALL query_rewrite.flush_rewrite_rules();
ERROR 45000: Loading of some rule(s) failed.
# We want the note about the query being rewritten here ...
SHOW WARNINGS;
Level	Code	Message
Error	1644	Loading of some rule(s) failed.
SHOW ERRORS;
Level	Code	Message
Error	1644	Loading of some rule(s) failed.
GET DIAGNOSTICS CONDITION 1 @p1 = RETURNED_SQLSTATE, @p2 = MESSAGE_TEXT;
SELECT @p1;
@p1
45000
SELECT @p2;
@p2
Loading of some rule(s) failed.
# ... but not here, since we failed parsing after seeing SHOW WARNINGS.
SHOW WARNINGS and give me a parse error, please;
ERROR 42000: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'and give me a parse error, please' at line 1
DELETE FROM query_rewrite.rewrite_rules;
# Test of warnings when we rewrite the diagnostics statement.  echo #
# We disable mtr's SHOW WARNINGS here, as it would just be confusing
# when it gets rewritten without us seeing it.
INSERT INTO query_rewrite.rewrite_rules ( pattern, replacement )
VALUES ( 'SHOW WARNINGS', 'SELECT "used to be SHOW WARNINGS"' ),
( 'SELECT 1', 'SHOW WARNINGS' );
CALL query_rewrite.flush_rewrite_rules();
ERROR 45000: Loading of some rule(s) failed.
SELECT 1;
Level	Code	Message
Error	1644	Loading of some rule(s) failed.
Note	1105	Query 'SELECT 1' rewritten to 'SHOW WARNINGS' by a query rewrite plugin
SHOW WARNINGS;
Level	Code	Message
Error	1644	Loading of some rule(s) failed.
Note	1105	Query 'SELECT 1' rewritten to 'SHOW WARNINGS' by a query rewrite plugin
Warnings:
Warning	1620	Plugin is busy and will be uninstalled on shutdown
# Query rewrite plugin was queued for uninstalling.
