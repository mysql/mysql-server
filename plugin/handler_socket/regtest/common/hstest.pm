
# vim:sw=2:ai

package hstest;

use DBI;
use Net::HandlerSocket;

our %conf = ();

sub get_conf_env {
  my ($key, $defval) = @_;
  return $ENV{$key} || $defval;
}

sub init_conf {
  $conf{host} = get_conf_env("MYHOST", "localhost");
  $conf{myport} = get_conf_env("MYPORT", 3306);
  $conf{dbname} = get_conf_env("MYDBNAME", "hstestdb");
  $conf{ssps} = get_conf_env("MYSSPS");
  $conf{user} = get_conf_env("MYSQLUSER", "root");
  $conf{pass} = get_conf_env("MYSQLPASS", "");
  $conf{hsport} = get_conf_env("HSPORT", 9998);
  $conf{hspass} = get_conf_env("HSPASS", undef);
}

sub get_dbi_connection {
  my ($dbname, $host, $myport, $ssps, $user, $pass)
    = ($conf{dbname}, $conf{host}, $conf{myport}, $conf{ssps},
      $conf{user}, $conf{pass});
  my $mycnf = "binary_my.cnf";
  my $dsn = "DBI:mysql:database=;host=$host;port=$myport"
    . ";mysql_server_prepare=$ssps"
    . ";mysql_read_default_group=perl"
    . ";mysql_read_default_file=../common/$mycnf";
  my $dbh = DBI->connect($dsn, $user, $pass, { RaiseError => 1 });
  return $dbh;
}

sub init_testdb {
  my $charset = $_[0] || "binary";
  my $dbh = get_dbi_connection();
  my $dbname = $conf{dbname};
  $dbh->do("drop database if exists $dbname");
  $dbh->do("create database $dbname default character set $charset");
  $dbh->do("use $dbname");
  return $dbh;
}

sub get_hs_connection {
  my ($host, $port) = @_;
  $host ||= $conf{host};
  $port ||= $conf{hsport};
  my $hsargs = { 'host' => $host, 'port' => $port };
  my $conn = new Net::HandlerSocket($hsargs);
  if (defined($conn) && defined($conf{hspass})) {
    $conn->auth($conf{hspass});
  }
  return $conn;
}


init_conf();

1;

