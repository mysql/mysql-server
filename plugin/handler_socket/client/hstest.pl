#!/usr/bin/perl

# vim:sw=8:ai:ts=8

use strict;
use warnings;

use DBI;
use Net::HandlerSocket;

my %conf = ();
for my $i (@ARGV) {
	my ($k, $v) = split(/=/, $i);
	$conf{$k} = $v;
}

my $verbose = get_conf("verbose", 0);
my $actions_str = get_conf("actions", "hsread");
my $tablesize = get_conf("tablesize", 10000);
my $db = get_conf("db", "hstest");
my $table = get_conf("table", "hstest_table1");
my $engine = get_conf("engine", "innodb");
my $host = get_conf("host", "localhost");
my $mysqlport = get_conf("mysqlport", 3306);
my $mysqluser = get_conf("mysqluser", "root");
my $mysqlpass = get_conf("mysqlpass", "");
my $hsport = get_conf("hsport", 9999);
my $loop = get_conf("loop", 10000);
my $op = get_conf("op", "=");
my $ssps = get_conf("ssps", 0);
my $num_moreflds = get_conf("moreflds", 0);
my $moreflds_prefix = get_conf("moreflds_prefix", "column0123456789_");
my $keytype = get_conf("keytype", "varchar(32)");
my $file = get_conf("file", undef);

my $dsn = "DBI:mysql:database=;host=$host;port=$mysqlport"
	. ";mysql_server_prepare=$ssps";
my $dbh = DBI->connect($dsn, $mysqluser, $mysqlpass, { RaiseError => 1 });
my $hsargs = { 'host' => $host, 'port' => $hsport };
my $cli = new Net::HandlerSocket($hsargs);

my @actions = split(/,/, $actions_str);
for my $action (@actions) {
	if ($action eq "table") {
		print("TABLE $db.$table\n");
		$dbh->do("drop database if exists $db");
		$dbh->do("create database $db");
		$dbh->do("use $db");
		my $moreflds = get_createtbl_moreflds_str();
		$dbh->do(
			"create table $table (" .
			"k $keytype primary key" .
			",v varchar(32) not null" .
			$moreflds .
			") character set utf8 collate utf8_bin " .
			"engine = $engine");
	} elsif ($action eq "insert") {
		print("INSERT $db.$table tablesize=$tablesize\n");
		$dbh->do("use $db");
		my $moreflds = get_insert_moreflds_str();
		for (my $i = 0; $i < $tablesize; $i += 100) {
			my $qstr = "insert into $db.$table values";
			for (my $j = 0; $j < 100; ++$j) {
				if ($j != 0) {
					$qstr .= ",";
				}
				my $k = "" . ($i + $j);
				my $v = "v" . int(rand(1000)) . ($i + $j);
				$qstr .= "('$k', '$v'";
				for (my $j = 0; $j < $num_moreflds; ++$j) {
					$qstr .= ",'$j'";
				}
				$qstr .= ")";
			}
			$dbh->do($qstr);
			print "$i/$tablesize\n" if $i % 1000 == 0;
		}
	} elsif ($action eq "read") {
		print("READ $db.$table op=$op loop=$loop\n");
		$dbh->do("use $db");
		my $moreflds = get_select_moreflds_str();
		my $sth = $dbh->prepare(
			"select k,v$moreflds from $db.$table where k = ?");
		for (my $i = 0; $i < $loop; ++$i) {
			my $k = "" . int(rand($tablesize));
			# print "k=$k\n";
			$sth->execute($k);
			if ($verbose >= 10) {
				print "RET:";
				while (my $ref = $sth->fetchrow_arrayref()) {
					my $rk = $ref->[0];
					my $rv = $ref->[1];
					print " $rk $rv";
				}
				print "\n";
			}
			print "$i/$loop\n" if $i % 1000 == 0;
		}
	} elsif ($action eq "hsinsert") {
		print("HSINSERT $db.$table tablesize=$tablesize\n");
		$cli->open_index(1, $db, $table, '', 'k,v');
		for (my $i = 0; $i < $tablesize; ++$i) {
			my $k = "" . $i;
			my $v = "v" . int(rand(1000)) . $i;
			my $r = $cli->execute_insert(1, [ $k, $v ]);
			if ($r->[0] != 0) {
				die;
			}
			print "$i/$tablesize\n" if $i % 1000 == 0;
		}
	} elsif ($action eq "hsread") {
		print("HSREAD $db.$table op=$op loop=$loop\n");
		my $moreflds = get_select_moreflds_str();
		$cli->open_index(1, $db, $table, '', "k,v$moreflds");
		for (my $i = 0; $i < $loop; ++$i) {
			my $k = "" . int(rand($tablesize));
			# print "k=$k\n";
			my $r = $cli->execute_find(1, $op, [ $k ], 1, 0);
			if ($verbose >= 10) {
				my $len = scalar(@{$r});
				print "LEN=$len";
				for my $e (@{$r}) {
					print " [$e]";
				}
				print "\n";
			}
			print "$i/$loop\n" if $i % 1000 == 0;
		}
	} elsif ($action eq "hsupdate") {
		my $vbase = "v" . int(rand(1000));
		print("HSUPDATE $db.$table op=$op loop=$loop vbase=$vbase\n");
		$cli->open_index(1, $db, $table, '', 'v');
		for (my $i = 0; $i < $loop; ++$i) {
			my $k = "" . int(rand($tablesize));
			my $v = $vbase . $i;
			print "k=$k v=$v\n";
			my $r = $cli->execute_update(1, $op, [ $k ], 1, 0,
				[ $v ]);
			if ($verbose >= 10) {
				print "UP k=$k v=$v\n";
			}
			print "$i/$loop\n" if $i % 1000 == 0;
		}
	} elsif ($action eq "hsdelete") {
		print("HSDELETE $db.$table op=$op loop=$loop\n");
		$cli->open_index(1, $db, $table, '', '');
		for (my $i = 0; $i < $loop; ++$i) {
			my $k = "" . int(rand($tablesize));
			print "k=$k\n";
			my $r = $cli->execute_delete(1, $op, [ $k ], 1, 0);
			if ($verbose >= 10) {
				print "DEL k=$k\n";
			}
			print "$i/$loop\n" if $i % 1000 == 0;
		}
	} elsif ($action eq "verify") {
		verify_do();
	}
}

sub verify_do {
	my ($fail_cnt, $ok_cnt) = (0, 0);
	my $sth = $dbh->prepare("select v from $db.$table where k = ?");
	use FileHandle;
	my $fh = new FileHandle($file, "r");
	while (my $line = <$fh>) {
		chomp($line);
		my @vec = split(/\t/, $line);
		my $k = $vec[3];
		my $v = $vec[7];
		next if (!defined($k) || !defined($v));
		# print "$k $v\n";
		$sth->execute($k);
		my $aref = $sth->fetchrow_arrayref();
		if (!defined($aref)) {
			print "FAILED: $k notfound\n";
			++$fail_cnt;
		} else {
			my $gv = $aref->[0];
			if ($gv ne $v) {
				print "FAILED: $k got=$gv expected=$v\n";
				++$fail_cnt;
			} else {
				print "OK: $k $v $gv\n" if $verbose >= 10;
				++$ok_cnt;
			}
		}
	}
	print "OK=$ok_cnt FAIL=$fail_cnt\n";
}

sub get_conf {
	my ($key, $def) = @_;
	my $val = $conf{$key};
	if ($val) {
		print "$key=$val\n";
	} else {
		$val = $def;
		$def ||= '';
		print "$key=$def(default)\n";
	}
	return $val;
}

sub get_createtbl_moreflds_str {
	my $s = "";
	for (my $j = 0; $j < $num_moreflds; ++$j) {
		$s .= ",$moreflds_prefix$j varchar(30)";
	}
	return $s;
}

sub get_select_moreflds_str {
	my $s = "";
	for (my $i = 0; $i < $num_moreflds; ++$i) {
		$s .= ",$moreflds_prefix$i";
	}
	return $s;
}

sub get_insert_moreflds_str {
	my $s = "";
	for (my $i = 0; $i < $num_moreflds; ++$i) {
		$s .= ",?";
	}
	return $s;
}

