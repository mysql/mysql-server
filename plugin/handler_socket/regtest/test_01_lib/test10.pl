#!/usr/bin/perl

# vim:sw=2:ai

# test for nulls

BEGIN {
	push @INC, "../common/";
};

use strict;
use warnings;
use hstest;

my $dbh = hstest::init_testdb();
my $table = 'hstesttbl';
my $tablesize = 256;
$dbh->do(
  "create table $table (" .
  "k varchar(30) primary key, " .
  "v1 varchar(30), " .
  "v2 varchar(30)) " .
  "engine = innodb default charset = binary");
srand(999);

my %valmap = ();

# setting null
print "HSINSERT";
my $hs = hstest::get_hs_connection(undef, 9999);
my $dbname = $hstest::conf{dbname};
$hs->open_index(1, $dbname, $table, '', 'k,v1,v2');
for (my $i = 0; $i < $tablesize; ++$i) {
  my $k = "" . $i;
  my $v1 = "v1_" . $i;
  my $v2 = undef; # null value
  my $r = $hs->execute_insert(1, [ $k, $v1, $v2 ]);
  my $err = $r->[0];
  if ($err != 0) {
    my $err_str = $r->[1];
    print "$err $err_str\n";
  }
}
undef $hs;

dump_table();

# setting null
print "HSUPDATE";
$hs = hstest::get_hs_connection(undef, 9999);
$dbname = $hstest::conf{dbname};
$hs->open_index(2, $dbname, $table, '', 'v1');
for (my $i = 0; $i < $tablesize; ++$i) {
  my $r = $hs->execute_single(2, '=', [ $i ], 1000, 0, 'U', [ undef ]);
  my $err = $r->[0];
  if ($err != 0) {
    my $err_str = $r->[1];
    print "$err $err_str\n";
  }
}
undef $hs;

dump_table();

# setting non-null
print "HSUPDATE";
$hs = hstest::get_hs_connection(undef, 9999);
$dbname = $hstest::conf{dbname};
$hs->open_index(2, $dbname, $table, '', 'v1');
for (my $i = 0; $i < $tablesize; ++$i) {
  my $r = $hs->execute_single(2, '=', [ $i ], 1000, 0, 'U', [ "hoge" ]);
  my $err = $r->[0];
  if ($err != 0) {
    my $err_str = $r->[1];
    print "$err $err_str\n";
  }
}
undef $hs;

dump_table();

sub dump_table {
  print "DUMP_TABLE\n";
  my $aref = $dbh->selectall_arrayref("select k,v1,v2 from $table order by k");
  for my $row (@$aref) {
    my ($k, $v1, $v2) = @$row;
    $v1 = "[null]" if !defined($v1);
    $v2 = "[null]" if !defined($v2);
    print "$k $v1 $v2\n";
    # print "MISMATCH\n" if ($valmap{$k} ne $v);
  }
}

