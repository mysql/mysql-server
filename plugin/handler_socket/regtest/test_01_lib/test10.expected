HSINSERTDUMP_TABLE
0 v1_0 [null]
1 v1_1 [null]
10 v1_10 [null]
100 v1_100 [null]
101 v1_101 [null]
102 v1_102 [null]
103 v1_103 [null]
104 v1_104 [null]
105 v1_105 [null]
106 v1_106 [null]
107 v1_107 [null]
108 v1_108 [null]
109 v1_109 [null]
11 v1_11 [null]
110 v1_110 [null]
111 v1_111 [null]
112 v1_112 [null]
113 v1_113 [null]
114 v1_114 [null]
115 v1_115 [null]
116 v1_116 [null]
117 v1_117 [null]
118 v1_118 [null]
119 v1_119 [null]
12 v1_12 [null]
120 v1_120 [null]
121 v1_121 [null]
122 v1_122 [null]
123 v1_123 [null]
124 v1_124 [null]
125 v1_125 [null]
126 v1_126 [null]
127 v1_127 [null]
128 v1_128 [null]
129 v1_129 [null]
13 v1_13 [null]
130 v1_130 [null]
131 v1_131 [null]
132 v1_132 [null]
133 v1_133 [null]
134 v1_134 [null]
135 v1_135 [null]
136 v1_136 [null]
137 v1_137 [null]
138 v1_138 [null]
139 v1_139 [null]
14 v1_14 [null]
140 v1_140 [null]
141 v1_141 [null]
142 v1_142 [null]
143 v1_143 [null]
144 v1_144 [null]
145 v1_145 [null]
146 v1_146 [null]
147 v1_147 [null]
148 v1_148 [null]
149 v1_149 [null]
15 v1_15 [null]
150 v1_150 [null]
151 v1_151 [null]
152 v1_152 [null]
153 v1_153 [null]
154 v1_154 [null]
155 v1_155 [null]
156 v1_156 [null]
157 v1_157 [null]
158 v1_158 [null]
159 v1_159 [null]
16 v1_16 [null]
160 v1_160 [null]
161 v1_161 [null]
162 v1_162 [null]
163 v1_163 [null]
164 v1_164 [null]
165 v1_165 [null]
166 v1_166 [null]
167 v1_167 [null]
168 v1_168 [null]
169 v1_169 [null]
17 v1_17 [null]
170 v1_170 [null]
171 v1_171 [null]
172 v1_172 [null]
173 v1_173 [null]
174 v1_174 [null]
175 v1_175 [null]
176 v1_176 [null]
177 v1_177 [null]
178 v1_178 [null]
179 v1_179 [null]
18 v1_18 [null]
180 v1_180 [null]
181 v1_181 [null]
182 v1_182 [null]
183 v1_183 [null]
184 v1_184 [null]
185 v1_185 [null]
186 v1_186 [null]
187 v1_187 [null]
188 v1_188 [null]
189 v1_189 [null]
19 v1_19 [null]
190 v1_190 [null]
191 v1_191 [null]
192 v1_192 [null]
193 v1_193 [null]
194 v1_194 [null]
195 v1_195 [null]
196 v1_196 [null]
197 v1_197 [null]
198 v1_198 [null]
199 v1_199 [null]
2 v1_2 [null]
20 v1_20 [null]
200 v1_200 [null]
201 v1_201 [null]
202 v1_202 [null]
203 v1_203 [null]
204 v1_204 [null]
205 v1_205 [null]
206 v1_206 [null]
207 v1_207 [null]
208 v1_208 [null]
209 v1_209 [null]
21 v1_21 [null]
210 v1_210 [null]
211 v1_211 [null]
212 v1_212 [null]
213 v1_213 [null]
214 v1_214 [null]
215 v1_215 [null]
216 v1_216 [null]
217 v1_217 [null]
218 v1_218 [null]
219 v1_219 [null]
22 v1_22 [null]
220 v1_220 [null]
221 v1_221 [null]
222 v1_222 [null]
223 v1_223 [null]
224 v1_224 [null]
225 v1_225 [null]
226 v1_226 [null]
227 v1_227 [null]
228 v1_228 [null]
229 v1_229 [null]
23 v1_23 [null]
230 v1_230 [null]
231 v1_231 [null]
232 v1_232 [null]
233 v1_233 [null]
234 v1_234 [null]
235 v1_235 [null]
236 v1_236 [null]
237 v1_237 [null]
238 v1_238 [null]
239 v1_239 [null]
24 v1_24 [null]
240 v1_240 [null]
241 v1_241 [null]
242 v1_242 [null]
243 v1_243 [null]
244 v1_244 [null]
245 v1_245 [null]
246 v1_246 [null]
247 v1_247 [null]
248 v1_248 [null]
249 v1_249 [null]
25 v1_25 [null]
250 v1_250 [null]
251 v1_251 [null]
252 v1_252 [null]
253 v1_253 [null]
254 v1_254 [null]
255 v1_255 [null]
26 v1_26 [null]
27 v1_27 [null]
28 v1_28 [null]
29 v1_29 [null]
3 v1_3 [null]
30 v1_30 [null]
31 v1_31 [null]
32 v1_32 [null]
33 v1_33 [null]
34 v1_34 [null]
35 v1_35 [null]
36 v1_36 [null]
37 v1_37 [null]
38 v1_38 [null]
39 v1_39 [null]
4 v1_4 [null]
40 v1_40 [null]
41 v1_41 [null]
42 v1_42 [null]
43 v1_43 [null]
44 v1_44 [null]
45 v1_45 [null]
46 v1_46 [null]
47 v1_47 [null]
48 v1_48 [null]
49 v1_49 [null]
5 v1_5 [null]
50 v1_50 [null]
51 v1_51 [null]
52 v1_52 [null]
53 v1_53 [null]
54 v1_54 [null]
55 v1_55 [null]
56 v1_56 [null]
57 v1_57 [null]
58 v1_58 [null]
59 v1_59 [null]
6 v1_6 [null]
60 v1_60 [null]
61 v1_61 [null]
62 v1_62 [null]
63 v1_63 [null]
64 v1_64 [null]
65 v1_65 [null]
66 v1_66 [null]
67 v1_67 [null]
68 v1_68 [null]
69 v1_69 [null]
7 v1_7 [null]
70 v1_70 [null]
71 v1_71 [null]
72 v1_72 [null]
73 v1_73 [null]
74 v1_74 [null]
75 v1_75 [null]
76 v1_76 [null]
77 v1_77 [null]
78 v1_78 [null]
79 v1_79 [null]
8 v1_8 [null]
80 v1_80 [null]
81 v1_81 [null]
82 v1_82 [null]
83 v1_83 [null]
84 v1_84 [null]
85 v1_85 [null]
86 v1_86 [null]
87 v1_87 [null]
88 v1_88 [null]
89 v1_89 [null]
9 v1_9 [null]
90 v1_90 [null]
91 v1_91 [null]
92 v1_92 [null]
93 v1_93 [null]
94 v1_94 [null]
95 v1_95 [null]
96 v1_96 [null]
97 v1_97 [null]
98 v1_98 [null]
99 v1_99 [null]
HSUPDATEDUMP_TABLE
0 [null] [null]
1 [null] [null]
10 [null] [null]
100 [null] [null]
101 [null] [null]
102 [null] [null]
103 [null] [null]
104 [null] [null]
105 [null] [null]
106 [null] [null]
107 [null] [null]
108 [null] [null]
109 [null] [null]
11 [null] [null]
110 [null] [null]
111 [null] [null]
112 [null] [null]
113 [null] [null]
114 [null] [null]
115 [null] [null]
116 [null] [null]
117 [null] [null]
118 [null] [null]
119 [null] [null]
12 [null] [null]
120 [null] [null]
121 [null] [null]
122 [null] [null]
123 [null] [null]
124 [null] [null]
125 [null] [null]
126 [null] [null]
127 [null] [null]
128 [null] [null]
129 [null] [null]
13 [null] [null]
130 [null] [null]
131 [null] [null]
132 [null] [null]
133 [null] [null]
134 [null] [null]
135 [null] [null]
136 [null] [null]
137 [null] [null]
138 [null] [null]
139 [null] [null]
14 [null] [null]
140 [null] [null]
141 [null] [null]
142 [null] [null]
143 [null] [null]
144 [null] [null]
145 [null] [null]
146 [null] [null]
147 [null] [null]
148 [null] [null]
149 [null] [null]
15 [null] [null]
150 [null] [null]
151 [null] [null]
152 [null] [null]
153 [null] [null]
154 [null] [null]
155 [null] [null]
156 [null] [null]
157 [null] [null]
158 [null] [null]
159 [null] [null]
16 [null] [null]
160 [null] [null]
161 [null] [null]
162 [null] [null]
163 [null] [null]
164 [null] [null]
165 [null] [null]
166 [null] [null]
167 [null] [null]
168 [null] [null]
169 [null] [null]
17 [null] [null]
170 [null] [null]
171 [null] [null]
172 [null] [null]
173 [null] [null]
174 [null] [null]
175 [null] [null]
176 [null] [null]
177 [null] [null]
178 [null] [null]
179 [null] [null]
18 [null] [null]
180 [null] [null]
181 [null] [null]
182 [null] [null]
183 [null] [null]
184 [null] [null]
185 [null] [null]
186 [null] [null]
187 [null] [null]
188 [null] [null]
189 [null] [null]
19 [null] [null]
190 [null] [null]
191 [null] [null]
192 [null] [null]
193 [null] [null]
194 [null] [null]
195 [null] [null]
196 [null] [null]
197 [null] [null]
198 [null] [null]
199 [null] [null]
2 [null] [null]
20 [null] [null]
200 [null] [null]
201 [null] [null]
202 [null] [null]
203 [null] [null]
204 [null] [null]
205 [null] [null]
206 [null] [null]
207 [null] [null]
208 [null] [null]
209 [null] [null]
21 [null] [null]
210 [null] [null]
211 [null] [null]
212 [null] [null]
213 [null] [null]
214 [null] [null]
215 [null] [null]
216 [null] [null]
217 [null] [null]
218 [null] [null]
219 [null] [null]
22 [null] [null]
220 [null] [null]
221 [null] [null]
222 [null] [null]
223 [null] [null]
224 [null] [null]
225 [null] [null]
226 [null] [null]
227 [null] [null]
228 [null] [null]
229 [null] [null]
23 [null] [null]
230 [null] [null]
231 [null] [null]
232 [null] [null]
233 [null] [null]
234 [null] [null]
235 [null] [null]
236 [null] [null]
237 [null] [null]
238 [null] [null]
239 [null] [null]
24 [null] [null]
240 [null] [null]
241 [null] [null]
242 [null] [null]
243 [null] [null]
244 [null] [null]
245 [null] [null]
246 [null] [null]
247 [null] [null]
248 [null] [null]
249 [null] [null]
25 [null] [null]
250 [null] [null]
251 [null] [null]
252 [null] [null]
253 [null] [null]
254 [null] [null]
255 [null] [null]
26 [null] [null]
27 [null] [null]
28 [null] [null]
29 [null] [null]
3 [null] [null]
30 [null] [null]
31 [null] [null]
32 [null] [null]
33 [null] [null]
34 [null] [null]
35 [null] [null]
36 [null] [null]
37 [null] [null]
38 [null] [null]
39 [null] [null]
4 [null] [null]
40 [null] [null]
41 [null] [null]
42 [null] [null]
43 [null] [null]
44 [null] [null]
45 [null] [null]
46 [null] [null]
47 [null] [null]
48 [null] [null]
49 [null] [null]
5 [null] [null]
50 [null] [null]
51 [null] [null]
52 [null] [null]
53 [null] [null]
54 [null] [null]
55 [null] [null]
56 [null] [null]
57 [null] [null]
58 [null] [null]
59 [null] [null]
6 [null] [null]
60 [null] [null]
61 [null] [null]
62 [null] [null]
63 [null] [null]
64 [null] [null]
65 [null] [null]
66 [null] [null]
67 [null] [null]
68 [null] [null]
69 [null] [null]
7 [null] [null]
70 [null] [null]
71 [null] [null]
72 [null] [null]
73 [null] [null]
74 [null] [null]
75 [null] [null]
76 [null] [null]
77 [null] [null]
78 [null] [null]
79 [null] [null]
8 [null] [null]
80 [null] [null]
81 [null] [null]
82 [null] [null]
83 [null] [null]
84 [null] [null]
85 [null] [null]
86 [null] [null]
87 [null] [null]
88 [null] [null]
89 [null] [null]
9 [null] [null]
90 [null] [null]
91 [null] [null]
92 [null] [null]
93 [null] [null]
94 [null] [null]
95 [null] [null]
96 [null] [null]
97 [null] [null]
98 [null] [null]
99 [null] [null]
HSUPDATEDUMP_TABLE
0 hoge [null]
1 hoge [null]
10 hoge [null]
100 hoge [null]
101 hoge [null]
102 hoge [null]
103 hoge [null]
104 hoge [null]
105 hoge [null]
106 hoge [null]
107 hoge [null]
108 hoge [null]
109 hoge [null]
11 hoge [null]
110 hoge [null]
111 hoge [null]
112 hoge [null]
113 hoge [null]
114 hoge [null]
115 hoge [null]
116 hoge [null]
117 hoge [null]
118 hoge [null]
119 hoge [null]
12 hoge [null]
120 hoge [null]
121 hoge [null]
122 hoge [null]
123 hoge [null]
124 hoge [null]
125 hoge [null]
126 hoge [null]
127 hoge [null]
128 hoge [null]
129 hoge [null]
13 hoge [null]
130 hoge [null]
131 hoge [null]
132 hoge [null]
133 hoge [null]
134 hoge [null]
135 hoge [null]
136 hoge [null]
137 hoge [null]
138 hoge [null]
139 hoge [null]
14 hoge [null]
140 hoge [null]
141 hoge [null]
142 hoge [null]
143 hoge [null]
144 hoge [null]
145 hoge [null]
146 hoge [null]
147 hoge [null]
148 hoge [null]
149 hoge [null]
15 hoge [null]
150 hoge [null]
151 hoge [null]
152 hoge [null]
153 hoge [null]
154 hoge [null]
155 hoge [null]
156 hoge [null]
157 hoge [null]
158 hoge [null]
159 hoge [null]
16 hoge [null]
160 hoge [null]
161 hoge [null]
162 hoge [null]
163 hoge [null]
164 hoge [null]
165 hoge [null]
166 hoge [null]
167 hoge [null]
168 hoge [null]
169 hoge [null]
17 hoge [null]
170 hoge [null]
171 hoge [null]
172 hoge [null]
173 hoge [null]
174 hoge [null]
175 hoge [null]
176 hoge [null]
177 hoge [null]
178 hoge [null]
179 hoge [null]
18 hoge [null]
180 hoge [null]
181 hoge [null]
182 hoge [null]
183 hoge [null]
184 hoge [null]
185 hoge [null]
186 hoge [null]
187 hoge [null]
188 hoge [null]
189 hoge [null]
19 hoge [null]
190 hoge [null]
191 hoge [null]
192 hoge [null]
193 hoge [null]
194 hoge [null]
195 hoge [null]
196 hoge [null]
197 hoge [null]
198 hoge [null]
199 hoge [null]
2 hoge [null]
20 hoge [null]
200 hoge [null]
201 hoge [null]
202 hoge [null]
203 hoge [null]
204 hoge [null]
205 hoge [null]
206 hoge [null]
207 hoge [null]
208 hoge [null]
209 hoge [null]
21 hoge [null]
210 hoge [null]
211 hoge [null]
212 hoge [null]
213 hoge [null]
214 hoge [null]
215 hoge [null]
216 hoge [null]
217 hoge [null]
218 hoge [null]
219 hoge [null]
22 hoge [null]
220 hoge [null]
221 hoge [null]
222 hoge [null]
223 hoge [null]
224 hoge [null]
225 hoge [null]
226 hoge [null]
227 hoge [null]
228 hoge [null]
229 hoge [null]
23 hoge [null]
230 hoge [null]
231 hoge [null]
232 hoge [null]
233 hoge [null]
234 hoge [null]
235 hoge [null]
236 hoge [null]
237 hoge [null]
238 hoge [null]
239 hoge [null]
24 hoge [null]
240 hoge [null]
241 hoge [null]
242 hoge [null]
243 hoge [null]
244 hoge [null]
245 hoge [null]
246 hoge [null]
247 hoge [null]
248 hoge [null]
249 hoge [null]
25 hoge [null]
250 hoge [null]
251 hoge [null]
252 hoge [null]
253 hoge [null]
254 hoge [null]
255 hoge [null]
26 hoge [null]
27 hoge [null]
28 hoge [null]
29 hoge [null]
3 hoge [null]
30 hoge [null]
31 hoge [null]
32 hoge [null]
33 hoge [null]
34 hoge [null]
35 hoge [null]
36 hoge [null]
37 hoge [null]
38 hoge [null]
39 hoge [null]
4 hoge [null]
40 hoge [null]
41 hoge [null]
42 hoge [null]
43 hoge [null]
44 hoge [null]
45 hoge [null]
46 hoge [null]
47 hoge [null]
48 hoge [null]
49 hoge [null]
5 hoge [null]
50 hoge [null]
51 hoge [null]
52 hoge [null]
53 hoge [null]
54 hoge [null]
55 hoge [null]
56 hoge [null]
57 hoge [null]
58 hoge [null]
59 hoge [null]
6 hoge [null]
60 hoge [null]
61 hoge [null]
62 hoge [null]
63 hoge [null]
64 hoge [null]
65 hoge [null]
66 hoge [null]
67 hoge [null]
68 hoge [null]
69 hoge [null]
7 hoge [null]
70 hoge [null]
71 hoge [null]
72 hoge [null]
73 hoge [null]
74 hoge [null]
75 hoge [null]
76 hoge [null]
77 hoge [null]
78 hoge [null]
79 hoge [null]
8 hoge [null]
80 hoge [null]
81 hoge [null]
82 hoge [null]
83 hoge [null]
84 hoge [null]
85 hoge [null]
86 hoge [null]
87 hoge [null]
88 hoge [null]
89 hoge [null]
9 hoge [null]
90 hoge [null]
91 hoge [null]
92 hoge [null]
93 hoge [null]
94 hoge [null]
95 hoge [null]
96 hoge [null]
97 hoge [null]
98 hoge [null]
99 hoge [null]
